examples/grocery_goaltask_img_ppo.gin

```diff
@@ -0,0 +1,84 @@
+# training & playing with Agent Learning Framework (Alf)
+# python -m alf.bin.train --root_dir=~/tmp/gro_img_ppo --gin_file=grocery_goaltask_img_ppo.gin --alsologtostderr
+# python -m alf.bin.play --root_dir=~/tmp/gro_img_ppo --gin_file=grocery_goaltask_img_ppo.gin
+
+import alf.algorithms.ppo_algorithm
+import alf.algorithms.ppo_loss
+import alf.algorithms.ddpg_algorithm
+import alf.trainers.off_policy_trainer
+
+TrainerConfig.trainer=@sync_off_policy_trainer
+TrainerConfig.algorithm_ctor=@create_ppo_algorithm
+ActorCriticAlgorithm.loss_class=@PPOLoss
+
+# environment config
+import alf.environments.wrappers
+CHANNEL_ORDER='channels_last'
+FrameStack.channel_order=%CHANNEL_ORDER
+GroceryGround.image_data_format=%CHANNEL_ORDER
+tf.keras.layers.Conv2D.data_format=%CHANNEL_ORDER
+create_environment.env_name='SocialBot-GroceryGroundImage-v0'
+create_environment.env_load_fn=@suite_socialbot.load
+suite_socialbot.load.gym_env_wrappers=(@FrameStack,)
+create_environment.num_parallel_environments=16
+GroceryGround.resized_image_size=(48, 48)
+
+PolicyDriver.observation_transformer=@image_scale_transformer
+
+
+# algorithm config
+ActorCriticAlgorithm.gradient_clipping=0.5
+ActorCriticAlgorithm.clip_by_global_norm=True
+ActorCriticAlgorithm.enforce_entropy_target=True
+
+PPOLoss.entropy_regularization=None
+PPOLoss.gamma=0.99
+PPOLoss.normalize_advantages=True
+PPOLoss.td_lambda=0.95
+PPOLoss.td_error_loss_fn=@element_wise_squared_loss
+
+# debug
+PPOLoss.check_numerics=True
+estimated_entropy.check_numerics=True
+
+import alf.networks.stable_normal_projection_network
+ActorDistributionNetwork.activation_fn=@tf.nn.tanh
+ActorDistributionNetwork.conv_layer_params=((16, 3, 2), (32, 3, 2))
+ActorDistributionNetwork.continuous_projection_net=@StableNormalProjectionNetwork
+StableNormalProjectionNetwork.init_means_output_factor=1e-10
+StableNormalProjectionNetwork.inverse_std_transform='softplus'
+StableNormalProjectionNetwork.scale_distribution=True
+StableNormalProjectionNetwork.state_dependent_std=True
+StableNormalProjectionNetwork.init_std=1.0
+
+# we need to modify network/stable_normal_projection_network.py line 125 to avoid action_log_prob NaN problem
+# Just setting the min/max does not help
+# inv_stds = self._std_transform(tf.clip_by_value(stds, 0.1, 100))
```

Comment:
Even add this clip it still reported this problem, just less unlikely. 
```
2019-09-18 19:43:32.164642: W tensorflow/core/common_runtime/base_collective_executor.cc:216] BaseCollectiveExecutor::StartAbort Invalid argument: action_log_prob : Tensor had NaN values
	 [[{{node while/body/_46/while/body/_725/CheckNumerics}}]]
Traceback (most recent call last):
  File "/usr/lib/python3.5/runpy.py", line 184, in _run_module_as_main
    "__main__", mod_spec)
  File "/usr/lib/python3.5/runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "/home/ubuntu/py35env/lib/python3.5/site-packages/tensorflow/python/eager/function.py", line 1324, in __call__
    return graph_function._filtered_call(args, kwargs)  # pylint: disable=protected-access
  File "/home/ubuntu/py35env/lib/python3.5/site-packages/tensorflow/python/eager/function.py", line 578, in _filtered_call
    (t for t in nest.flatten((args, kwargs), expand_composites=True)
  File "/home/ubuntu/py35env/lib/python3.5/site-packages/tensorflow/python/eager/function.py", line 660, in _call_flat
    outputs = self._inference_function.call(ctx, args)
  File "/home/ubuntu/py35env/lib/python3.5/site-packages/tensorflow/python/eager/function.py", line 434, in call
    ctx=ctx)
  File "/home/ubuntu/py35env/lib/python3.5/site-packages/tensorflow/python/eager/execute.py", line 67, in quick_execute
    six.raise_from(core._status_to_exception(e.code, message), None)
  File "<string>", line 3, in raise_from
tensorflow.python.framework.errors_impl.InvalidArgumentError: action_log_prob : Tensor had NaN values
	 [[{{node while/body/_46/while/body/_725/CheckNumerics}}]] [Op:__inference_train_55000]
  In call to configurable 'train_eval' (<function train_eval at 0x7f033a95c950>)

```

Reply:
Another fix I have to make things more stable for scale_distribution is the following. I didn't add it to ALF because I filed a PR to tf_agents and it was merged to tf_agents (https://github.com/tensorflow/agents/pull/166) . But you need to update to a newer version of agents to use it. Or you need to do the following patch.
```python
from tf_agents.distributions import tanh_bijector_stable


def _patched_inverse_tanh(self, y):
    # 0.99999997 is the maximum value such that atanh(x) is valid for both
    # tf.float32 and tf.float64
    y = tf.where(
        tf.less_equal(tf.abs(y), 1.),
        tf.clip_by_value(y, -0.99999997, 0.99999997), y)

    return tf.atanh(y)


# Make Tanh bijector more numerically stable. Because of limited floating point
# precision, tanh(9.01) will get 1.0, which cannot be inverted. An input value
# of 9.01 is not the uncommon. Consider a Gaussian distribution with mean=1.0
# and std=2. You can get a sample larger than 9.01 or smaller than -9.01 with
# probability 6.2e-5. So we change Tanh._inverse() to make it stabler.
tanh_bijector_stable.Tanh._inverse = _patched_inverse_tanh

```

Reply:
This solves the problem.

Reply:
Still have a chance reporting error of "action_log_prob : Tensor had NaN values"

Reply:
You can try to log action distribution and the action used for calculating action_log_prob when NaN happens to find out what is the cause.